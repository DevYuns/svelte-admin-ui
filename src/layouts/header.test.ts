import Header from './header.svelte';
import {getStorage} from '../utils/storage';
import {render} from '@testing-library/svelte';
import {setUpUser} from './../../test/testUtils';

describe('render test', () => {
    it('render without crashing', () => {
        const {container} = render(Header);

        expect(container).toMatchSnapshot();
    });
});

describe('interaction test', () => {
    it('should log out when a user click logout button', async () => {
        const {getAllByRole, user} = setUpUser(Header);

        const logoutATag = getAllByRole('link')[2];

        await user.click(logoutATag);

        expect(getStorage('token')).toBeNull();
    });
});
