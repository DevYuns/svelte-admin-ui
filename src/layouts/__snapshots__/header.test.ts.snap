// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test render without crashing 1`] = `
<body>
  <div>
    <nav
      class="header svelte-5agz0t"
    >
      <a
        class="logo svelte-5agz0t"
        href="mockURL"
      >
        Svelte-Admin-UI
      </a>
       
      <ul
        class="svelte-5agz0t"
      >
        <li>
          <a
            class="link svelte-5agz0t active"
            href="mockURL"
          >
            Settings
          </a>
        </li>
         
        <li>
          <a
            class="link svelte-5agz0t"
            href=""
          >
            logout
          </a>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
